modules/non_summary_pipeline.py

```diff
@@ -417,6 +418,7 @@ def __init__(
         dataset_requests_home_only,
         dataset_response_bodies_home_only,
         label=None,
+        triggering_frequency=None,
```

Comment:
Should there be a default if this is needed to make the pipeline work?

Reply:
No need, passing `None` all the way through to will cause the pipeline to use beam's own defaults for `WriteToBigQuery`.

https://beam.apache.org/releases/pydoc/2.40.0/apache_beam.io.gcp.bigquery.html?highlight=writetobigquery#apache_beam.io.gcp.bigquery.WriteToBigQuery

> triggering_frequency ([float](https://docs.python.org/3/library/functions.html#float)) –
When method is FILE_LOADS: Value will be converted to int. Every triggering_frequency seconds, a BigQuery load job will be triggered for all the data written since the last load job. BigQuery has limits on how many load jobs can be triggered per day, so be careful not to set this duration too low, or you may exceed daily quota. Often this is set to 5 or 10 minutes to ensure that the project stays well under the BigQuery quota. See https://cloud.google.com/bigquery/quota-policy for more information about BigQuery quotas.
> 
> When method is STREAMING_INSERTS and with_auto_sharding=True: A streaming inserts batch will be submitted at least every triggering_frequency seconds when data is waiting. The batch can be sent earlier if it reaches the maximum batch size set by batch_size. Default value is 0.2 seconds.

Reply:
Going to double check this by adding a unit test.

Reply:
Nope I was wrong. Checked the SDK code and there is no default.

We will need to avoid going over the quota for load jobs, currently 100,000 jobs per day.
https://cloud.google.com/bigquery/quotas#load_jobs

As an example, the validation job above has been running for about 4 hours and generated 2,333 load jobs with `triggering_frequency=15`. If left running for 24 hours, this job would produce about 14,000 jobs, which is well under the daily quota.

Adding a default of 15 minutes.

Reply:
Is the unit minutes or seconds? The help docs above talk about "every triggering_frequency seconds".

Reply:
Shoot... yes, the unit is definitely seconds. The default value we provide should be 900 (i.e. 15 * 50). Updated now 🤦